Bulk pull ascending

clemahieu opened the pull request.

jamescoxon committed:
upgraded Qt to 6 (originally released in 2020), while the last version of Qt5 will have support for 3 years for commercial licence users increasingly there will be a move to Qt6 - for example for M1 silicon only Qt6 is available (for now https://www.qt.io/blog/qt-on-apple-silicon)
also important to recognise that nano_wallet is rarely used these days due to better community gui wallets

jamescoxon committed:
ran cmake_format

jamescoxon committed:
added back compatibility to Qt5 (5.15)

jamescoxon committed:
ran cmake_format again

dsiganos commented:
The protocol docs and kaitai systax will also need updating.
And possibly the protocol version too.

clemahieu committed:
Adding bootstrap_ascending bootstrapping mode.

clemahieu committed:
Extract probability transform and documenting current backoff function.

clemahieu committed:
Adding perf test for bootstrap

clemahieu committed:
WIP

First pass at moving the backoff weights table on to disk. This mapping grows proportionally to the number of accounts and needs to persist between runs.

mdb_iterator is used for its ability to find the-next-key after a value.
mdb_get/put are used for direct searching and inserting.

dsiganos commented:
This has been superseded by Piotr's version of the ascending bootstrap.
